src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1291,7 +1316,30 @@ private void registerKeyStoreProviderOnConnection(String keyStoreAuth, String ke
                         systemColumnEncryptionKeyStoreProvider.put(provider.getName(), provider);
                     }
                     break;
-
+                case KeyVaultClientSecret:
+                    // need a secret use use the secret method
+                    if (null == keyStoreSecret) {
+                        throw new SQLServerException(
+                                SQLServerException.getErrString("R_keyStoreSecretOrLocationNotSet"), null);
+                    } else {
+                        SQLServerColumnEncryptionAzureKeyVaultProvider provider = new SQLServerColumnEncryptionAzureKeyVaultProvider(
+                                keyStorePrincipalId, keyStoreSecret);
+                        Map<String, SQLServerColumnEncryptionKeyStoreProvider> keyStoreMap = new HashMap<String, SQLServerColumnEncryptionKeyStoreProvider>();
+                        keyStoreMap.put(provider.getName(), provider);
+                        registerColumnEncryptionKeyStoreProviders(keyStoreMap);
```

Comment:
Feels like a waste to create a hash-map w/ 1 entry just to conform to the existing constructor. Maybe this can be done better?